OPHD/MicroPather/micropather.cpp

```diff
@@ -294,15 +294,20 @@ void PathNodePool::Clear()
 
 PathNodePool::Block* PathNodePool::NewBlock()
 {
-	Block* block = reinterpret_cast<Block*>(calloc(1, sizeof(Block) + sizeof(PathNode) * (allocate - 1)));
-	block->nextBlock = nullptr;
-
-	nAvailable += allocate;
-	for (std::size_t i = 0; i < allocate; ++i)
+	// assign-then-check result is non-nullptr
+	if(Block* block = reinterpret_cast<Block*>(calloc(1, sizeof(Block) + sizeof(PathNode) * (allocate - 1))))

```

Comment:
I think I would prefer the assignment be left outside of the `if`. Stylistically, I would like to avoid assignments within `if` expressions. Sometimes it makes sense to do, but I don't think it's needed here.

Leaving it outside would probably also make the diff look a bit cleaner.

Reply:
I disagree. This is a well-known C++ idiom as well as it keeps the scope of `block` as small as possible to let the compiler optimizer do its thing. Some tools will even warn `block variable scope can be reduced`. Objectively all a good thing.

Reply:
Yes, though some compilers will also warn about assignment in `if` statements. They are very easily misread, and in many cases done by accident. This case seems a little better, since the variable is being declared, not simply assigned.

As for block scope optimizations, can you name any that would apply? I'm not aware of any optimizations that would kick in for this particular case.

Reply:
They don't warn for pointers.

`if(ptr)` never warns.

Reply:
It does warn for `if (ptr = expression)`.

Reply:
Yes, but that's not what `if(type* ptr = expression)` is. It's a declaration, not an assignment.